test/tstOBB.cpp

```diff
@@ -0,0 +1,194 @@
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX_OBB.hpp>
+
+#include "BoostTest_CUDA_clang_workarounds.hpp"
+#include <boost/test/unit_test.hpp>
+
+#include <tuple>
+#include <type_traits>
+
+namespace tt = boost::test_tools;
+
+using ArborX::Experimental::OBB;
+
+BOOST_AUTO_TEST_SUITE(OrientedBoundingBox)
+
+BOOST_AUTO_TEST_CASE(equals_obb)
+{
+  using ArborX::Details::equals;
+
+  OBB<2, float> obb({{1, 0}, {0, 1}});
+  BOOST_TEST(equals(obb, obb));
+
+  OBB<2, float> obb1({{1, 0}, {0, 1}, {0, 0}});
+  BOOST_TEST(!equals(obb, obb1));
+
+  OBB<3, float> obb3({{1, 0, 0}, {0, 1, 0}, {0, 0, 1}});
+  BOOST_TEST(equals(obb3, obb3));
+}
+
+BOOST_AUTO_TEST_CASE(expand_obb_2D)
+{
+  using Point = ArborX::ExperimentalHyperGeometry::Point<2>;
+  using ArborX::Details::equals;
+  using ArborX::Details::expand;
+
+  Point points[3] = {{1, 0}, {0, 0}, {0, 1}};
+
+  OBB<2, float> obb;
+  expand(obb, points[0]);
+  BOOST_TEST(equals(obb, OBB({points[0]})));
+
+  expand(obb, OBB({points[1], points[2]}));
+  BOOST_TEST(equals(obb, OBB({points[0], points[1], points[2]})));
+}
+
+BOOST_AUTO_TEST_CASE(expand_obb_3D)
+{
+  using Point = ArborX::ExperimentalHyperGeometry::Point<3>;
+  using ArborX::Details::equals;
+  using ArborX::Details::expand;
+
+  Point points[5] = {{1, 0, 0}, {0, 0, 0}, {0, 2, 0}, {0, 0, 3}, {0, 2, 3}};
+
+  OBB<3, float> obb;
+  expand(obb, points[0]);
+  BOOST_TEST(equals(obb, OBB({points[0]})));
+
+  expand(obb, OBB({points[1], points[2], points[3], points[4]}));
+  BOOST_TEST(equals(
+      obb, OBB({points[0], points[1], points[2], points[3], points[4]})));
+}
+
+BOOST_AUTO_TEST_CASE(intersects_point_obb_2D)
+{
+  using Point = ArborX::ExperimentalHyperGeometry::Point<2>;
+  using ArborX::Details::intersects;
+
+  {
+    OBB<2, float> obb({{1, 0}});
+    BOOST_TEST(intersects(Point{1, 0}, obb));
+    BOOST_TEST(!intersects(Point{0, 0}, obb));
+  }
+  {
+    //   x
+    //    \
+    //     \
+    //      x
+    OBB<2, float> obb({{1, 0}, {0, 1}});
+    BOOST_TEST(intersects(Point{0.5f, 0.5f}, obb));
+    BOOST_TEST(intersects(Point{0, 1}, obb));
+    BOOST_TEST(intersects(Point{1, 0}, obb));
+    BOOST_TEST(!intersects(Point{0, 0}, obb));
+  }
+  {
+    //   x
+    //  / \
+    //  \  \
+    //   x  x
+    //    \/
+    OBB<2, float> obb({{1, 0}, {0, 1}, {0, 0}});
+    BOOST_TEST(intersects(Point{-0.25f, 0.26f}, obb));
+    BOOST_TEST(!intersects(Point{-0.5f, 0.49f}, obb));
+    BOOST_TEST(intersects(Point{-0.25f, 0.74f}, obb));
+    BOOST_TEST(intersects(Point{0.25f, -0.24f}, obb));
+    BOOST_TEST(!intersects(Point{0.5f, 0.51f}, obb));
+    BOOST_TEST(intersects(Point{0.75f, 0.24f}, obb));
+  }
+  {
+    // OBB is not necessarily the tightest box around the points. Here's an
+    // example where the covariance matrix is proportional to identity, and
+    // we simply choose the default axes, i.e.,
```

Comment:
Do you know why that's the case?

Reply:
My understanding that this is due to the sampling bias. Corners of a polytope may not represent it sufficiently well from the probabilistic sampling point of view. See also Sections 3.3.1 and 3.3.2 of the Gottschalk's [dissertation](https://gamma.cs.unc.edu/users/gottschalk/main.pdf).